### DISCUSSION TITLE
Cast value before perform syncInput

### DISCUSSION THREAD

**Discussion by ph7jack** [livewire/livewire#3370] [3. Feature ideas]
I need a way to cast value before performing the update.

Like the livewire life cycle, **hydratingFoo**, in this method, cast the variable and return the new value

```
public function hydratingFoo($value) 
{
      return cast($value);
}
```

**Comment by joshhanley** [comment]
@PH7-Jack do you mean cast the original value before the updated value gets applied, or cast the new updated value before it gets applied?

If it's the later, try using updating hook and pass value by reference `&`

```php
public function updatingFoo(&$value)
{
    $value = cast($value);
}
```

I think that should work.

Hope this helps!

**Reply by ph7jack** [reply]
Yes, cast the value before the update.

I tried using **updatingFoo**, but the value is applied before call this hook. 
Is a typed variable that's accept Carbon or null, set string values will broken
I need cast it before set

**Reply by joshhanley** [reply]
@PH7-Jack yeah you need to pass $value by reference, notice the `&$value` in the method signature.

Give that a try, and let me know how you go.

If you cant get it to work, I can have a look tomorrow, as I'm sure I've done it before using the pass by reference like my example here.

**Reply by ph7jack** [reply]
Good, this work, thanks

```
public function updatingDate(&$val)
{
    $val = Carbon::parse($val);

    // this don't need
    $this->date = $val;
}
```

**Reply by Lukmauu** [reply]
@joshhanley you deserve a medal sir, thank you for this answer.

**Reply by joshhanley** [reply]
@Lukmauu no worries! Glad it helped! 🙂

**Reply by Jamiul-Bari** [reply]
@joshhanley  Does it work on Livewire 3?

I am facing this on livewire 3 where I cast a numeric input into float using `updatingFoo(&$value)` but in updated() hook it says

`Property [$amount] not found on component:`

**Reply by joshhanley** [reply]
@Jamiul-Bari It should do. But I think that error is unrelated. It sounds like you don't have a single root element in your component.

**Reply by Roardom** [reply]
Unfortunately this does not currently work in Livewire v3: https://github.com/livewire/livewire/pull/8061

Is there any alternative?

**Reply by joshhanley** [reply]
Yeah it turns out this doesn't work in V3 and isn't a simple thing to change. The best thing to do would be to just make the changes in an `updated` hook or have a look at building custom synths to handle property types in custom ways https://livewire.laravel.com/docs/synthesizers

**Comment by trippo** [comment]
Work like a charme
